Soccar [Game-Server]/src/nl/soccar/gameserver/SessionController.java

```diff
@@ -61,128 +58,88 @@ public boolean createSession(String name, String password, String hostName, int
         gameSettings.setMapType(mapType);
         gameSettings.setBallType(ballType);
 
+        SessionWrapper wrapper = new SessionWrapper(server, this, session);
+
         synchronized (sessions) {
-            sessions.add(session);
+            sessions.put(name, wrapper);
             controller.sessionCreated(name, hostName, !password.isEmpty(), capacity);
         }
 
         LOGGER.log(Level.INFO, "Session {0} created.", name);
         return true;
     }
 
-    public void destroySession(String name) {
+    public void destroySession(Session session) {
+        if (session == null) {
+            return;
+        }
+
+        String name = session.getRoom().getName();
         synchronized (sessions) {
-            Optional<Session> optional = sessions.stream().filter(s -> s.getRoom().getName().equals(name)).findAny();
-            if (optional.isPresent()) {
-                sessions.remove(optional.get());
+            SessionWrapper wrapper = sessions.remove(name);
+            if (wrapper == null) {
+                return;
             }
+
+            // TODO remove all players, or something...

```

Comment:
![INFO](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-info.png) Complete the task associated to this TODO comment. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://194.135.91.67:9000/coding_rules#rule_key=squid%3AS1135)